src/daemon/inout_route.rs

```diff
@@ -103,3 +90,29 @@ pub async fn out_route_obfsudp(
         .await;
     }
 }
+
+async fn per_route_tasks(
+    ctx: DaemonContext,
+    pipe: impl Pipe,
+    their_fp: Option<Fingerprint>,
+) -> anyhow::Result<()> {
+    log::info!("about to connect link_conn");
+
+    let link = LinkConnection::connect(ctx.clone(), pipe, their_fp).await?;
```

Comment:
the connection_task should not be part of LinkConnection anymore --- that's a pretty important part of the refactor.

instead, we should start the task here. LinkConnection should be an "inert" object.

this also seems to be the correct place to insert something into NeighTable --- probably the sending end of a channel, not a LinkConnection object (LinkConnection can probably be gotten rid of)